src/mrinufft/operators/base.py

```diff
@@ -139,6 +139,13 @@ def wrapper(self, data, output=None, *args, **kwargs):
             data_ = data
             output_ = output
 
+        if output_ is not None:
+            if not((
+                is_host_array(data_) and is_host_array(output_)
+            ) or (
+                is_cuda_array(data_) and is_cuda_array(output_))
+            ):
```

Comment:
I added these lines here. I think it is a good check. For now we dont support CPU / GPU input outputs